lib/RallyValidate.js

```diff
@@ -60,21 +60,41 @@ class RallyValidate {
       }
     }
 
-    const rallyClient = rally({
-      user: rallyUsername, // Required if no api key, defaults to process.env.RALLY_USERNAME
-      pass: rallyPassword, // Required if no api key, defaults to process.env.RALLY_PASSWORD
-      apiKey: rallyAPIKey, // Preferred, required if no user/pass, defaults to process.env.RALLY_API_KEY
-      apiVersion: 'v2.0',  // This is the default and may be omitted
-      server: rallyServer, // This is the default and may be omitted
-      requestOptions: {
-        headers: {
-          'X-RallyIntegrationName': 'Rally + GitHub', // while optional, it is good practice to
-          'X-RallyIntegrationVendor': 'GitHub, Inc',  // provide this header information
-          'X-RallyIntegrationVersion': '1.0'
+    let rallyClient
+
+    if (process.env.RALLY_PROXY !== undefined) {
+      rallyClient = rally({
+        user: rallyUsername, // Required if no api key, defaults to process.env.RALLY_USERNAME
+        pass: rallyPassword, // Required if no api key, defaults to process.env.RALLY_PASSWORD
+        apiKey: rallyAPIKey, // Preferred, required if no user/pass, defaults to process.env.RALLY_API_KEY
+        apiVersion: 'v2.0',  // This is the default and may be omitted
+        server: rallyServer, // This is the default and may be omitted
+        requestOptions: {
+          headers: {
+            'X-RallyIntegrationName': 'Rally + GitHub', // while optional, it is good practice to
+            'X-RallyIntegrationVendor': 'GitHub, Inc',  // provide this header information
+            'X-RallyIntegrationVersion': '1.0'
+          },
+          proxy: process.env.RALLY_PROXY
         }
-        // any additional request options (proxy options, timeouts, etc.)
-      }
-    })
+      })
+    } else {
+      rallyClient = rally({
+        user: rallyUsername, // Required if no api key, defaults to process.env.RALLY_USERNAME
+        pass: rallyPassword, // Required if no api key, defaults to process.env.RALLY_PASSWORD
+        apiKey: rallyAPIKey, // Preferred, required if no user/pass, defaults to process.env.RALLY_API_KEY
+        apiVersion: 'v2.0',  // This is the default and may be omitted
+        server: rallyServer, // This is the default and may be omitted
+        requestOptions: {
+          headers: {
+            'X-RallyIntegrationName': 'Rally + GitHub', // while optional, it is good practice to
+            'X-RallyIntegrationVendor': 'GitHub, Inc',  // provide this header information
+            'X-RallyIntegrationVersion': '1.0'
+          }
+          // any additional request options (proxy options, timeouts, etc.)
+        }
+      })
+    }
```

Comment:
Hey @migarjo!

It might be better to do something like that. What do you think?

```js
const rallyClient = rally({
  user: rallyUsername, // Required if no api key, defaults to process.env.RALLY_USERNAME
  pass: rallyPassword, // Required if no api key, defaults to process.env.RALLY_PASSWORD
  apiKey: rallyAPIKey, // Preferred, required if no user/pass, defaults to process.env.RALLY_API_KEY
  apiVersion: 'v2.0',  // This is the default and may be omitted
  server: rallyServer, // This is the default and may be omitted
  requestOptions: {
    headers: {
      'X-RallyIntegrationName': 'Rally + GitHub', // while optional, it is good practice to
      'X-RallyIntegrationVendor': 'GitHub, Inc',  // provide this header information
      'X-RallyIntegrationVersion': '1.0'
    },
    ...(process.env.RALLY_PROXY !== undefined && { proxy: process.env.RALLY_PROXY })
    // any additional request options (proxy options, timeouts, etc.)
  }
})
```

Reply:
Thanks for the suggestion, @joaocarvalhowd! I agree, this is a much cleaner approach. I'll make that adjustment.